client/X11/xf_event.c

```diff
@@ -477,6 +477,9 @@ static BOOL xf_event_KeyRelease(xfContext* xfc, XEvent* event, BOOL app)
 }
 static BOOL xf_event_FocusIn(xfContext* xfc, XEvent* event, BOOL app)
 {
+	if (!xfc || !event || !xfc->window)
+		return FALSE;
```

Comment:
how the hell can we have `xfc == NULL` or `event == NULL`, if this happens I guess this is a problem with the caller (static function). Even the check on `xfc->window` is questionable.